src/rule_definition_tools.jl

```diff
@@ -56,6 +56,8 @@ derivative/setup expressions.
 This macro assumes complex functions are holomorphic. In general, for non-holomorphic
 functions, the `frule` and `rrule` must be defined manually.
 
+Prefer using `true` over `1.0` to express multiplicative identity.
```

Comment:
```suggestion
If the derivative is one, (e.g. for identity functions) `true` can be used as the most general multiplicative identity.
```

Reply:
I did some benchmarking and looking `@code_native` and found that for floating point input they all generate the same code

```julia
julia> f(x) = true*x
f (generic function with 1 method)

julia> f(x) = true*x;

julia> g(x) = 1 * x;

julia> h(x) = 1.0 * x;

julia> j(x) = x;
```
that code being
```
julia> @code_native j(1.4)
        .section        __TEXT,__text,regular,pure_instructions
; ┌ @ REPL[28]:1 within `j`
        retq
        nopw    %cs:(%rax,%rax)
; └
```

So maybe it doesn't really matter at all.

Reply:
Similarly

```julia
julia> @scalar_rule muladd(x, y, z) (y, x, true)

julia> @btime y, pb = rrule(muladd, 1.0, 2.0, 3.0); pb(1)
  0.034 ns (0 allocations: 0 bytes)
(ZeroTangent(), 2, 1, 1)

julia> @scalar_rule muladd(x, y, z) (y, x, One())

julia> @btime y, pb = rrule(muladd, 1.0, 2.0, 3.0); pb(1)
  0.033 ns (0 allocations: 0 bytes)
(ZeroTangent(), 2, 1, 1)
```